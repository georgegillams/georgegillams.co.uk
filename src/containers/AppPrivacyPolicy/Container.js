import React from 'react';
import PageContainer from 'components/common/PageContainer';
import PageTitle from 'components/common/PageTitle';
import Paragraph from '@george-gillams/components/paragraph';

import Subsection from '@george-gillams/components/subsection';
import { withScrollAnimation } from '@george-gillams/components/effects';
const SubsectionWithScroll = withScrollAnimation(Subsection);

const AppPrivacyPolicy = () => {
  return (
    <PageContainer bottomPadding>
      <PageTitle name="App privacy policy">
        <SubsectionWithScroll name="Data and analytics">
          <Paragraph>
            My apps don&#39;t collect any data from your device. Any information the apps save is stored locally on your
            device. They don&#39;t even collect any analytics. I may add analytics to my apps in the future to
            understand how they are used and help prioritise features, but I will never collect or sell your personal
            data.
          </Paragraph>
        </SubsectionWithScroll>
        <SubsectionWithScroll name="Cookies">
          <Paragraph>
            None of my apps have any reason to store cookies or session information. So they don&#39;t.
          </Paragraph>
        </SubsectionWithScroll>
        <SubsectionWithScroll name="Version">
          <Paragraph>This is version 1 of my app privacy policy.</Paragraph>
        </SubsectionWithScroll>
      </PageTitle>
    </PageContainer>
  );
};

export default AppPrivacyPolicy;
