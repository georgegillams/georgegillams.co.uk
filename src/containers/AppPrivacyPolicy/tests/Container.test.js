import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import configureStore from 'client-utils/common/redux/configure-store';

import AppPrivacyPolicy from '../Container';

describe('<AppPrivacyPolicy />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render correctly - index', () => {
    const { container } = render(
      <Provider store={store}>
        <AppPrivacyPolicy />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
