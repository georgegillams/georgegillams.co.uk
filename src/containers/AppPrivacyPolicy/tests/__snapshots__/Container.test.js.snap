// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppPrivacyPolicy /> should render correctly - index 1`] = `
.c0 {
  display: block;
  width: 100%;
  margin-right: auto;
  margin-left: auto;
  overflow: hidden;
  padding-bottom: 2rem;
  max-width: 60rem;
  padding-right: 1rem;
  padding-left: 1rem;
}

.c2 {
  margin: 0;
  font-size: 2.4rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  font-size: 1.5rem;
  font-weight: bold;
}

.c3 {
  margin: 2.3rem 0 0 0;
}

.c5 {
  padding-top: 0;
  opacity: 1;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c4 {
  width: 100%;
}

.c1 {
  -webkit-animation: bbwGtI 0.4s ease-in-out;
  animation: bbwGtI 0.4s ease-in-out;
}

.c9 {
  display: inline-block;
  margin-bottom: 0;
  padding-bottom: 0;
  line-height: 1.5;
  margin-top: 2rem;
}

.c9:first-of-type {
  margin-top: 0.86rem;
}

.c6 {
  width: 100%;
  color: #1e1e1e;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 {
  margin: 2.13rem 0 0 0;
}

@media (min-width:50.25rem) {
  .c0 {
    padding-right: 2rem;
    padding-left: 2rem;
  }
}

@media (prefers-color-scheme:dark) {
  .c6 {
    color: white;
  }
}

<div>
  <div
    class="c0 page-containerstyles__StyledPageContainer-sc-19l7k4-0"
    width="default"
  >
    <div
      class="c1"
    >
      <h1
        class="c2 c3"
      >
        App privacy policy
      </h1>
    </div>
    <div
      class="c4"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="c5 animated-content__wrapper--hidden"
      >
        <div
          class="c6"
        >
          <h1
            class="c7 c8"
            id="data-and-analytics"
          >
            Data and analytics
          </h1>
          <span
            class="c9"
          >
            My apps don't collect any data from your device. Any information the apps save is stored locally on your device. They don't even collect any analytics. I may add analytics to my apps in the future to understand how they are used and help prioritise features, but I will never collect or sell your personal data.
          </span>
        </div>
      </div>
    </div>
    <div
      class="c4"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="c5 animated-content__wrapper--hidden"
      >
        <div
          class="c6"
        >
          <h1
            class="c7 c8"
            id="cookies"
          >
            Cookies
          </h1>
          <span
            class="c9"
          >
            None of my apps have any reason to store cookies or session information. So they don't.
          </span>
        </div>
      </div>
    </div>
    <div
      class="c4"
    >
      <style>
        .js .animated-content__wrapper--hidden {
  padding-top: 4rem;
  opacity: 0.01;
}
      </style>
      <div
        class="c5 animated-content__wrapper--hidden"
      >
        <div
          class="c6"
        >
          <h1
            class="c7 c8"
            id="version"
          >
            Version
          </h1>
          <span
            class="c9"
          >
            This is version 1 of my app privacy policy.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
